core/test/junit/saros/git/GitManagerTest.java

```diff
@@ -0,0 +1,239 @@
+import saros.activities.IActivity;
+import saros.net.xmpp.JID;
+import saros.session.IActivityConsumer;
+import saros.session.IActivityListener;
+import saros.session.User;
+import saros.session.internal.SarosSession;
+
+@RunWith(PowerMockRunner.class)
+@PrepareForTest(SarosSession.class)
+public class GitManagerTest {
+
+  private SarosSession alicesSession;
+  private IActivityConsumer alicesConsumer;
+  private User alicesAlice;
+  private User alicesBob;
+
+  private SarosSession bobsSession;
+  private IActivityConsumer bobsConsumer;
+  private User bobsAlice;
+  private User bobsBob;
+
+  @Before
+  public void createSessionsMocks() {
+    alicesSession = PowerMock.createNiceMock(SarosSession.class);
+    alicesSession.addActivityProducer(isA(GitManager.class));
+    alicesSession.removeActivityProducer(isA(GitManager.class));
+
+    alicesSession.addActivityConsumer(
+        isA(IActivityConsumer.class), isA(IActivityConsumer.Priority.class));
+    EasyMock.expectLastCall()
+        .andStubAnswer(
+            new IAnswer<Object>() {
+              @Override
+              public Object answer() throws Throwable {
+                alicesConsumer = (IActivityConsumer) EasyMock.getCurrentArguments()[0];
+                return null;
+              }
+            });
+    alicesSession.removeActivityConsumer(isA(IActivityConsumer.class));
+
+    alicesAlice = new User(new JID("alice"), true, true, 1, -1);
+    alicesBob = new User(new JID("bob"), false, false, 2, -1);
+
+    alicesAlice.setInSession(true);
+    alicesBob.setInSession(true);
+
+    alicesSession.getLocalUser();
+    EasyMock.expectLastCall().andStubReturn(alicesAlice);
+
+    EasyMock.replay(alicesSession);
+
+    bobsSession = PowerMock.createNiceMock(SarosSession.class);
+
+    bobsSession.addActivityProducer(isA(GitManager.class));
+    bobsSession.removeActivityProducer(isA(GitManager.class));
+
+    bobsSession.addActivityConsumer(
+        isA(IActivityConsumer.class), isA(IActivityConsumer.Priority.class));
+    EasyMock.expectLastCall()
+        .andStubAnswer(
+            new IAnswer<Object>() {
+              @Override
+              public Object answer() throws Throwable {
+                bobsConsumer = (IActivityConsumer) EasyMock.getCurrentArguments()[0];
+                return null;
+              }
+            });
+    bobsSession.removeActivityConsumer(isA(IActivityConsumer.class));
+
+    bobsAlice = new User(new JID("alice"), true, false, 1, -1);
+    bobsBob = new User(new JID("bob"), false, true, 2, -1);
+
+    bobsAlice.setInSession(true);
+    bobsBob.setInSession(true);
+
+    bobsSession.getLocalUser();
+    EasyMock.expectLastCall().andStubReturn(bobsBob);
+
+    EasyMock.replay(bobsSession);
+  }
+
+  @Test
+  public void testCreation() {
+    GitManager gitManager = new GitManager(alicesSession);
+    gitManager.start();
+    gitManager.stop();
+    EasyMock.verify(alicesSession);
+  }
+
+  @Test
```

Comment:
This test does not test anything.

Reply:
To write a test for a manager I get inspired by [StopManagerTest.java](https://github.com/saros-project/saros/blob/master/core/test/junit/saros/synchronize/StopManagerTest.java). I used the test I have seen there to assert in my test `Assert.assertEquals(alicesBob, activity.getSource());`